### DISCUSSION TITLE
Variables (names) got semantic token type `function`

### DISCUSSION THREAD

**Discussion by yttriumz** [BetterThanTomorrow/calva#2666] [Q&A]
Hi! I wonder why variables (names) are identified as `function`, thus rendered in the same color as functions. The name bindings should be different from functions. (Correct me if I'm wrong. I'm new to Clojure :)

For example in the following screenshots (using Inspect Editor Tokens and Scopes):

function:

<img src="https://github.com/user-attachments/assets/28e8ee47-cfdd-4720-835e-fd0cb9c20ba3" alt="demo" width="60%"/>

name:

<img src="https://github.com/user-attachments/assets/68b1bdfd-ea37-44fd-b7dc-cc5269a281f7" alt="demo" width="60%"/>

**Comment by PEZ** [comment]
Hi! The semantic tokens are provided by clojure-lsp. So maybe @ericdallo can shed some light on the reasoning around the tokens chosen.

**Comment by ericdallo** [comment]
From clojure POV, those things are `var`s, specifically `var-definitions` from clj-kondo analysis, so they are the same, they can be used as functions and the content can only be known in repl, since clojure-lsp use static analysis, it can't differ those things, even so, I don't see any problem with current behavior, they are highlighted as functions because the LSP spec doesn't have anything more specific too
